mock/py/mockbuild/mounts.py

```diff
@@ -224,13 +224,13 @@ def add_user_mount(self, mount):
 
     @traceLog()
     def mountall_essential(self):
+        self.essential_mounted = True
         for m in self.essential_mounts:
             m.mount()
-        self.essential_mounted = True
 
     @traceLog()
     def mountall_managed(self):
-        if not util.USE_NSPAWN:
+        if not self.essential_mounted and not util.USE_NSPAWN:
```

Comment:
Hm, thinking again ... it shouldn't be insecure to call mountall_essential twice.  Can this hunk be dropped?

Reply:
I wasn't suggesting this was insecure.

If we reach this and `self.essential_mounted` is set here, then we know everything in `self.essential_mounts` was successfully mounted by mock. What would be the benefit of looping over the list again?

If any of the failed to mount, it would have raised an exception, and mock would have unmounted them and exited.

Reply:
What I dislike is that the check is not encapsulated.  **All the callers** of `mount_essential` have to do this check ... which means that we either should move the check into `mountall_essential`, or leave it as is.

And actually, the list of essential mounts is dynamic ... so subsequent calls to mountall_essential is expected.

Reply:
> And actually, the list of essential mounts is dynamic ... so subsequent calls to mountall_essential is expected.

Ok, that part I missed. Then this should be removed.

Reply:
Can you please update this?